### DISCUSSION TITLE
How to manage the open source parts of twinBASIC

### DISCUSSION THREAD

**Discussion by mansellan** [twinbasic/twinbasic#838] [General]
Certain parts of twinBASIC are intended to be open source:

- The custom IDE
- The VS Code extension (paused)
- Language design discussions?
- Documentation?
- Sample packages?

Open sourcing raises various questions:

- Where will the sources be placed? GitHub, or elsewhere?
- What is the process to accept changes? Pull requests? If so, should there be a reviewer pool?
- What license will be used? MIT / Apache / GPL / Custom?
- What happens if licenses need to change?
  - This can happen accidentally, if an essential component needs to be brought in that has an incompatible license
  - Before accepting pull requests, a contributer license should be required that allows license change. 

Legals are a minefield;d.  I would recommend these questions are answered before any open source repos are created.

**Comment by DinyaZ** [comment]
-Documentation for example, and community members from other countries can help to create local translations. It is easy, and almost every applications uses this.
- and it would be very useful if the community create a sample code pool with free, smaller or bigger demo projects. Good for us to learn, but good for the project also, because this shows a friendly message to the possible newcomers, and they can choose the TWB easier with seemingly better suppor. This will build the community (and Wayne's bank account :) ). But a stronger community is a big benefit to us also, because with more contributors the project can be more mature, so win-win.

-IDE is also interesting subject, because the community can perhaps develop other, custum built IDE versions
